XMonad/Doc/Extending.hs

```diff
@@ -349,13 +347,13 @@ provide useful functions for editing your key bindings; "XMonad.Util.EZConfig" e
 {- $keyDel
 #Removing_key_bindings#
 
-Removing key bindings requires modifying the 'Data.Map.Map' which
-stores the key bindings.  This can be done with 'Data.Map.difference'
-or with 'Data.Map.delete'.
+Removing key bindings requires modifying the 'Data.Map.Strict.Map' which
+stores the key bindings.  This can be done with 'Data.Map.difference' or
+with 'Data.Map.Strict.delete'.
```

Comment:
Hm, this is really unfortunate. :-/

> The Map type is shared between the lazy and strict modules, meaning that the same Map value can be passed to functions in both modules. This means that the Functor, Traversable and Data instances are the same as for the Data.Map.Lazy module, so if they are used the resulting maps may contain suspended values (thunks).

(https://hackage.haskell.org/package/containers-0.6.5.1/docs/Data-Map-Strict.html)

So we really need to link to `Data.Map.Strict.Map` for haddock to produce a good link, but we also don't really want to confuse users into importing `Data.Map.Strict` instead of `Data.Map`. :-/

And then these links don't work with xmonad-docs anyway, as the base haddocks aren't there. :-/

There's probably no good way to do this.

Reply:
Users importing `Data.Map.Strict` isn't so bad, actually.  The base data structures (`Data.Map.Lazy.Map` and `Data.Map.Strict.Map`, that is) are actually the same; the functions in the stricd module just make sure to force things before handling them.  I don't think we're really relying on the lazyness of `Data.Map` anywhere in xmonad.  The fact that the links don't work, however, is not good :/

Reply:
True, it's not bad. I did find it confusing as the code uses `Data.Map` everywhere. Maybe it doesn't matter.

Fixing the links may involve some `sed` hackery and can be somewhat brittle, so I'd deprioritize it somehow (I do need a break, after all).

Reply:
I mean, there's nothing stopping us from rewriting the docs (and the code, eventually) to use `Data.Map.Strict` instead of `Data.Map`

Reply:
Which won't fix the links. ;-)